src/python/pants/backend/awslambda/python/targets.py

```diff
@@ -6,15 +6,28 @@
 
 
 class PythonAwsLambdaHandler(StringField):
-    """Lambda handler entrypoint (module.dotted.name:handler_func)."""
+    """AWS Lambda handler entrypoint (module.dotted.name:handler_func)."""
 
     alias = "handler"
     required = True
     value: str
 
 
+class PythonAwsLambdaRuntime(StringField):
+    """The identifier of the AWS Lambda runtime to target (pythonX.Y)."""
+
+    alias = "runtime"
+    required = True
+    value: str
```

Comment:
If you want to stick with `StringField`, you can use `valid_choices = ("python3.7", "python3.8")` etc for simpler validation and for better documentation with `target-types2 --details=python_awslambda`.

Reply:
I prefer not to have valid_choices, because we'll need code changes (and a forced upgrade) when AWS introduce a "python3.9" runtime etc.